plugins/proxy-sigv4-backend/src/service/router.ts

```diff
@@ -168,31 +178,46 @@ export function buildMiddleware(
   ) => {
     try {
       const requestHeaders = filterHeaders(req.headers as HeadersMap);
-
-      const requestOptions: any = {
-        method: req.method as any,
-        searchParams: req.query as any,
-        headers: requestHeaders as any,
+      const targetUrl = new URL(req.url, target);
+      requestHeaders.host = targetUrl.host;
+
+      const request: any = {
+        method: req.method,
+        protocol: targetUrl.protocol,
+        hostname: targetUrl.host,
```

Comment:
unused?

Reply:
good eye! This is likely a remnant from a previous iteration. I should be able to clean this up.

Reply:
Resolved in https://github.com/segmentio/segment-backstage-plugins/pull/10/commits/66252fa42e72685398118a5811ad649d061e4cfa